### DISCUSSION TITLE
Adaptive formatting for memory units (bytes)

### DISCUSSION THREAD

**Discussion by mkopec87** [apache/superset#30558] [Ideas]
Memory unit (bytes) is typically formatted into kB (kilobytes), MB (megabytes), etc.
Superset formatting options don't enable adaptive formatting with proper unit naming in case of bytes values.

**Comment by mkopec87** [comment]
I have created a PR with this feature: https://github.com/apache/superset/pull/30559

**Comment by mkopec87** [comment]
Merged to master

**Comment by Minimalistikaa** [comment]
Hey,
Thanks for that feature.
Sadly i cannot access it in the dropdown menu in "Y Axis Format"
I checked if my superset version has the file in "superset/superset-frontend/packages/superset-ui-core/src/number-format/factories/createMemoryFormatter.ts" and it indeed exists.

I am new to superset, do i need to activate this formatter first so i can use it in my chard editor?

Greets

**Reply by mkopec87** [reply]
Hi! Which version of the superset you're running? Do you have the commit hash of your version to share?
And how did you start your instance?

I didn't need to activate anything while I was testing the feature, just ran the docker compose up and everything worked out of the box, but maybe someone changed something in the meantime.

**Reply by Minimalistikaa** [reply]
Hey, thanks for your anwser.

I start superset with "superset run -h 0.0.0.0 -p 8088 --with-threads --reload --debugger" at the moment as we still evaluate the software.

> screen -S superset
source superset-venv/bin/activate
export SUPERSET_CONFIG_PATH=~/.superset/superset_config.py
export FLASK_APP=superset
superset run -h 0.0.0.0 -p 8088 --with-threads --reload --debugger

It has been installed it with "pip install apache-superset" in a virtual enviroment. So i should have the official version.
The Version seems to be "4.0.2"

**Reply by mkopec87** [reply]
Thanks for sharing the info! Version 4.0.2 will not have this feature as it was added later than 4.0.2 was released. :( 

In fact, this feature is so new that it hasn't been included in any official release yet.
From the comment in here: https://github.com/apache/superset/pull/30559#issuecomment-2483646592 I understand that it will appear in version 4.2 or 5.0 (whichever comes first).

It's still possible to test that feature by running the version from source (from master branch).

**Reply by Minimalistikaa** [reply]
Thank you,
That explains that. :)

I guess we wait for the update then as it is just a cosmetic problem for now.
